src/tests/ftest/util/nvme_utils.py

```diff
@@ -38,6 +40,69 @@ def get_device_ids(test, dmg, servers):
     return devices
 
 
+def set_device_faulty(test, dmg, server, uuid, pool=None, **kwargs):
+    """Set the device faulty and optionally wait for rebuild to complete.
+
+    Args:
+        test (Test): avocado test class
+        dmg (DmgCommand): a DmgCommand class instance
+        server (NodeSet): host on which to issue the dmg storage set nvme-faulty
+        uuid (str): the device UUID
+        pool (TestPool, optional): pool used to wait for rebuild to start/complete if specified.
+            Defaults to None.
+
+    Returns:
+        CmdResult: _description_
+
+    """
+    dmg.hostlist = server
+    kwargs['uuid'] = uuid
+    dmg_json = dmg.json.value
```

Comment:
Let's keep the comments in the code I sent to allow for a easy edit when `storage_set_faulty()` supports json output:
```suggestion
    # Use this when DmgCommand.storage_set_faulty returns json output natively
    # response = get_dmg_response(dmg.storage_set_faulty, **kwargs)

    # Remove this when DmgCommand.storage_set_faulty returns json output natively
    dmg_json = dmg.json.value
```